File: 065.LeetCode@61_Rotate_List/065.LeetCode@61_Rotate_List.py
class Solution:
    def rotateRight(self, head, k: int):
        if not head or k == 0 or not head.next:
            return head
        temp = head
        count = 0
        while temp is not None:
            if temp.next is not None:
                temp = temp.next
                count += 1
            else:
                end = temp
                count += 1
                break
        if count == k or (k % count == 0 and k != 1):
            return head
        temp = head
        loc = (count + k) % count
        for i in range(count - loc - 1):
            temp = temp.next
        start = temp.next
        temp.next = None
        end.next = head
        return start

File: 065.LeetCode@61_Rotate_List/065.LeetCode@61_Rotate_List.java
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        if (head == null || k == 0 || head.next == null) {
            return head;
        }
        ListNode temp = head;
        int count = 0;
        ListNode end= head;
        while (temp != null) {
            end = temp;
            temp = temp.next;
            count += 1;
        }
        if (count == k || (k % count == 0 && k != 1)) {
            return head;
        }
        temp = head;
        int loc = (count + k) % count;
        for (int i = 0; i < count - loc -1; i++) {
            temp = temp.next;
        }
        ListNode start = temp.next;
        temp.next = null;
        end.next = head;
        return start;
    }
}